planner/core/planbuilder.go

```diff
@@ -1760,6 +1754,57 @@ func (b *PlanBuilder) buildExplain(explain *ast.ExplainStmt) (Plan, error) {
 			return nil, ErrUnsupportedType.GenWithStackByArgs(targetPlan)
 		}
 	}
+	return pp, nil
+}
+
+// buildExplainFor gets *last* (maybe running or finished) query plan from connection #connection id.
+// See https://dev.mysql.com/doc/refman/8.0/en/explain-for-connection.html.
+func (b *PlanBuilder) buildExplainFor(explainFor *ast.ExplainForStmt) (Plan, error) {
+	processInfo, ok := b.ctx.GetSessionManager().ShowProcessList()[explainFor.ConnectionID]
```

Comment:
`b.ctx.GetSessionManager().ShowProcessList()` constructs a `map[uint64]util.ProcessInfo` and returns it, this operation is too heavy to be used in this context.

We can add another specific function in `SessionManager` interface to get the process info for a connection. For example `GetProcessInfo(conn int) util.ProcessInfo`